lib/bedrock.js

```diff
@@ -320,6 +243,59 @@ api.exit = function() {
   cluster.worker.kill();
 };
 
+async function _runOnce({type, id, fn, options}) {
+  // listen for scheduling message from master
+  const promise = _waitForOneMessage({type, id});
+
+  // notify master to schedule work (if not already scheduled/run)
+  process.send({type, id, options});
+
+  // wait for scheduling result
+  let msg = await promise;
+
+  // work completed in another worker, finish
+  if(msg.done) {
+    if(msg.error) {
+      throw errio.fromObject(msg.error, {stack: true});
+    }
+    return;
+  }
+
+  // run in this worker
+  msg = {type, id, done: true};
+  let error;
+  try {
+    await fn();
+  } catch(e) {
+    error = e;
+    msg.error = cycle.decycle(errio.toObject(e, {stack: true}));
+  }
+
+  // notify other workers that work is complete
+  process.send(msg);
+
+  if(error) {
+    throw error;
+  }
+}
+
+function _waitForOneMessage({type, id}) {
+  // get coordinated message from master
+  const promise = new Promise(resolve => {
```

Comment:
this could be just `return new Promise(...)` ?

Also, is there a convention to have promise returning things be `async function ...`?

Reply:
Yup, simplified.